src/nupic/algorithms/backtracking_tm.py

```diff
@@ -574,7 +573,10 @@ def read(cls, proto):
     obj.pamCounter = int(proto.pamCounter)
     obj.collectSequenceStats = proto.collectSequenceStats
     obj.resetCalled = proto.resetCalled
-    obj.avgInputDensity = proto.avgInputDensity
+    if proto.avgInputDensity < 0.0:
```

Comment:
Also, not sure if we care in this case, but I think I've seen recommendations to access proto members as few times as possible for performance reasons (I think it parses each time). In typical case, we'll read `proto.avgInputDensity` first to check for negative, then second time to assign it to `obj.avgInputDensity`. IIRC, the recommended practice was to save the value in a variable and then compare, assign, etc.